eng/common/TestResources/New-TestResources.ps1

```diff
@@ -140,15 +156,18 @@ try {
     # Enumerate test resources to deploy. Fail if none found.
     $repositoryRoot = "$PSScriptRoot/../../.." | Resolve-Path
     $root = [System.IO.Path]::Combine($repositoryRoot, "sdk", $ServiceDirectory) | Resolve-Path
-    $templateFileName = 'test-resources.json'
     $templateFiles = @()
 
-    Write-Verbose "Checking for '$templateFileName' files under '$root'"
-    Get-ChildItem -Path $root -Filter $templateFileName -Recurse | ForEach-Object {
-        $templateFile = $_.FullName
-
-        Write-Verbose "Found template '$templateFile'"
-        $templateFiles += $templateFile
+    'test-resources.json', 'test-resources.bicep' | ForEach-Object {
+        Write-Verbose "Checking for '$_' files under '$root'"
+        Get-ChildItem -Path $root -Filter "$_" -Recurse | ForEach-Object {
+            Write-Verbose "Found template '$($_.FullName)'"
+            if ($_.Extension -eq '.bicep' -and !(Get-Command bicep -ErrorAction Ignore)) {
```

Comment:
Any reason to not move this command check into your help function and also do the bicep build at this point in the execution?

Reply:
My thinking at the time was following the gather/collect/build stages that the script has, but I think it can probably be reasonably simplified now that I've added some other logic resulting from the PR discussion.